docs/book/src/clusterctl/configuration.md

```diff
@@ -48,3 +49,41 @@ AWS_B64ENCODED_CREDENTIALS: XXXXXXXX
 ```
 
 In case a variable is defined both in the config file and as an OS environment variable, the latter takes precedence.
+
+## Image overrides
+
+<aside class="note warning">
+
+<h1> Warning! </h1>
+
+Image override is an advanced feature and wrong configuration can easily lead to non-functional clusters.
+It's strongly recommended to test configurations on dev/test environments before using this functionality in production.
+
+</aside>
+
+When working in air-gapped environments, it's necessary to alter the manifests to be installed in order to pull
+images from a local/custom image repository instead of public ones (e.g. `gcr.io`, or `quay.io`).
+
+The `clusterctl` configuration file can be used to instruct `clusterctl` to override images automatically.
+
+This can be achieved by adding an `images` configuration entry as shown in the example:
+
+```yaml
+images:
+  all:
+    repository: myorg.io/local-repo
+```
+
+Please note that the image override feature allows for more fine-grained configuration, allowing to set image
+overrides for specific components, for example:
+
+```yaml
+images:
+  all:
+    repository: myorg.io/local-repo
+  cert-manager:
```

Comment:
can all component images be overridden? If not, would it make sense to provide a list of images that can be?

Reply:
@CecileRobertMichon 
ATM we are not supporting overrides for a single image, but only overrides at component level. In this example, "cert-manager" means all the images in the cert-manager manifest.

Available components are "cert-manger", "cluster-api" and all the installed providers: "bootstrap-kubeadm", "control-plane-kubeadm", "infrastracture-aws" etc.

Eventually, in the future/if there are specific needs, the mechanism can be extended for implementing image specific settings, but at the end the list of images depends on the provider manifests so it will be up to the provides to document which images are used.